### PR TITLE
Fix segfault crash from bundle IDs

### PR DESCRIPTION
**Pull request by betamos**
This fixes a segfault/crash in #8 

There are some other QOL improvements as well. If you aren't convinced by them, we can break them out and fix the critical issue first.

### CODE REVIEW

**Comment by lucasfernog** [comment]
I can confirm this works and i know how to reproduce this in tauri if you guys need a way to test it.

File: tests/application.rs [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 
 #[test]
-#[should_panic]
 fn set_application_again() {
-    let _ = set_application("com.apple.Terminal").unwrap();
-    let _ = set_application("com.apple.Terminal").unwrap();
+    let mut result = set_application("com.apple.Terminal");
+    assert!(result.is_ok());
+    result = set_application("com.apple.Terminal");
+    assert!(result.is_err());
```
Comment (hoodie) [inline tests/application.rs:8]: nitpick: unwrapping this first `result` actually preferred - generally nicer debug experience should it actually be an `Err`

**Comment by hoodie** [comment]
looks really! thanks for the effort! out of curiosity, I'd love to see a reproducer for that segfault somehow. maybe you could add that as an example.

also: we don't mention this in the README yet, but I'd love if you could prefix the commit messages with `fix:` and `refactor:` (a.k.a [convention commit messages](https://www.conventionalcommits.org/)) because I use [convco](https://github.com/convco/convco) to generate the changelog.

File: src/lib.rs [hunk 2]
```diff
@@ -101,26 +98,32 @@ pub fn get_bundle_identifier_or_default(app_name: &str) -> String {
...
+    INIT_APPLICATION_SET.call_once(|| {
+        let was_set = unsafe { sys::setApplication(NSString::from_str(bundle_ident).deref()) };
+        result = if was_set {
+            Ok(())
+        } else {
+            Err(ApplicationError::CouldNotSet(bundle_ident.into()).into())
+        };
+    });
+
+    return result;
```
[the remark this answers is not in the document]
Comment (betamos) [reply src/lib.rs]: Done

File: tests/application.rs (hunk 1 above)
Comment (betamos) [reply tests/application.rs]: Good point!

**Review by betamos — COMMENTED** [review]
Thanks! Addressed your comments.

**Comment by betamos** [comment]
> I'd love to see a reproducer for that segfault somehow

It's virtually impossible to get a reliable repro, but for me the `simple.rs` example did the trick eventually. I'm on aarch64 (M1) with MacOS 12.3.1. There linked issue has some stack traces and build options for debug info.

Oh, and could you please bump the crate version post-merge?

**Comment by hoodie** [comment]
et voila https://crates.io/crates/mac-notification-sys/0.5.1

**Pull request merged**